These tests are failing:
```python
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModel::test_enable_free_onset_energy
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModel::test_disable_free_onset_energy
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModel::test_lazy_enable_free_onset_energy
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModel::test_lazy_disable_free_onset_energy
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModelFitting::test_enable_free_onset_energy_fitting_test
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModelFitting::test_disable_free_onset_energy_fitting_test
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModelFitting::test_lazy_enable_free_onset_energy_fitting_test
FAILED hyperspy/tests/model/test_eelsmodel.py::TestEELSModelFitting::test_lazy_disable_free_onset_energy_fitting_test
```

--- a/hyperspy/models/eelsmodel.py
+++ b/hyperspy/models/eelsmodel.py
@@ -902,7 +902,13 @@ class EELSModel(Model1D):
             edges_list = [self._get_component(x) for x in edges_list]
         for edge in edges_list:
             if edge.isbackground is False:
-                edge.intensity.free = True
+                edge.set_parameters_free(
+                    parameter_name_list=[
+                        "intensity",
+                        "onset_energy",
+                        "fine_structure_coeff",
+                    ]
+                )
 
     def fix_fine_structure(self, edges_list=None):
         """Fixes the fine structure of the edges given in edges_list.

--- a/hyperspy/models/eelsmodel.py
+++ b/hyperspy/models/eelsmodel.py
@@ -149,7 +149,7 @@ class EELSModel(Model1D):
         """Classify components between background and ionization edge
         components.
 
-        This method should be called everytime that components are added and
+        This method should be called every time that components are added and
         removed. An ionization edge becomes background when its onset falls to
         the left of the first non-masked energy channel. The ionization edges
         are stored in a list in the `edges` attribute. They are sorted by
@@ -195,7 +195,8 @@ class EELSModel(Model1D):
         return [bc for bc in self._background_components if bc.active]
 
     def _add_edges_from_subshells_names(self, e_shells=None):
-        """Create the Edge instances and configure them appropiately
+        """Create the Edge instances and configure them appropriately
+
         Parameters
         ----------
         e_shells : list of strings
@@ -243,7 +244,7 @@ class EELSModel(Model1D):
             i1=0):
         """Adjust the fine structure of all edges to avoid overlapping
 
-        This function is called automatically everytime the position of an edge
+        This function is called automatically every time the position of an edge
         changes
 
         Parameters
@@ -348,8 +349,8 @@ class EELSModel(Model1D):
         the energy-loss-axis. First it fits only the background
         up to the first edge. It continues by deactivating all
         edges except the first one, then performs the fit. Then
-        it only activates the the first two, fits, and repeats
-        this until all edges are fitted simultanously.
+        it only activates the first two, fits, and repeats
+        this until all edges are fitted simultaneously.
 
         Other, non-EELSCLEdge components, are never deactivated,
         and fitted on every iteration.
@@ -421,7 +422,7 @@ class EELSModel(Model1D):
 
         """
 
-        # If there is no active background compenent do nothing
+        # If there is no active background component do nothing
         if not self._active_background_components:
             return
         iee = self._get_first_ionization_edge_energy(start_energy=start_energy)
@@ -591,7 +592,7 @@ class EELSModel(Model1D):
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -623,7 +624,7 @@ class EELSModel(Model1D):
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -654,7 +655,7 @@ class EELSModel(Model1D):
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -678,7 +679,7 @@ class EELSModel(Model1D):
         self.resolve_fine_structure()
 
     def enable_background(self):
-        """Enable the background componets.
+        """Enable the background components.
 
         """
         for component in self._background_components:
@@ -698,7 +699,7 @@ class EELSModel(Model1D):
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -729,7 +730,7 @@ class EELSModel(Model1D):
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -785,11 +786,11 @@ class EELSModel(Model1D):
         """Enable the automatic freeing of the onset_energy parameter during a
         smart fit for the edges listed in edges_list.
         If edges_list is None (default) the onset_energy of all the edges
-        with onset in the spectrum energy region will be freeed.
+        with onset in the spectrum energy region will be freed.
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -816,12 +817,12 @@ class EELSModel(Model1D):
         smart fit for the edges listed in edges_list.
         If edges_list is None (default) the onset_energy of all the edges
         with onset in the spectrum energy region will not be freed.
-        Note that if their atribute edge.onset_energy.free is True, the
+        Note that if their attribute edge.onset_energy.free is True, the
         parameter will be free during the smart fit.
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -850,7 +851,7 @@ class EELSModel(Model1D):
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -876,11 +877,11 @@ class EELSModel(Model1D):
 
     def free_edges(self, edges_list=None):
         """Frees all the parameters of the edges given in edges_list.
-        If edges_list is None (default) all the edges will be freeed.
+        If edges_list is None (default) all the edges will be freed.
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -904,12 +905,12 @@ class EELSModel(Model1D):
                 edge.intensity.free = True
 
     def fix_fine_structure(self, edges_list=None):
-        """Fixes all the parameters of the edges given in edges_list.
+        """Fixes the fine structure of the edges given in edges_list.
         If edges_list is None (default) all the edges will be fixed.
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -932,12 +933,12 @@ class EELSModel(Model1D):
                 edge.fine_structure_coeff.free = False
 
     def free_fine_structure(self, edges_list=None):
-        """Frees all the parameters of the edges given in edges_list.
-        If edges_list is None (default) all the edges will be freeed.
+        """Frees the fine structure of the edges given in edges_list.
+        If edges_list is None (default) all the edges will be freed.
 
         Parameters
         ----------
-        edges_list : None or  list of EELSCLEdge or list of edge names
+        edges_list : None or list of EELSCLEdge or list of edge names
             If None, the operation is performed on all the edges in the model.
             Otherwise, it will be performed only on the listed components.
 
@@ -960,7 +961,7 @@ class EELSModel(Model1D):
                 edge.fine_structure_coeff.free = True
 
     def suspend_auto_fine_structure_width(self):
-        """Disable the automatic adjustament of the core-loss edges fine
+        """Disable the automatic adjustment of the core-loss edges fine
         structure width.
 
         See Also
@@ -974,14 +975,13 @@ class EELSModel(Model1D):
             warnings.warn("Already suspended, does nothing.")
 
     def resume_auto_fine_structure_width(self, update=True):
-        """Enable the automatic adjustament of the core-loss edges fine
+        """Enable the automatic adjustment of the core-loss edges fine
         structure width.
 
         Parameters
         ----------
         update : bool, optional
             If True, also execute the automatic adjustment (default).
-
 
         See Also
         --------

--- a/hyperspy/tests/model/test_eelsmodel.py
+++ b/hyperspy/tests/model/test_eelsmodel.py
@@ -100,6 +100,7 @@ class TestEELSModel:
         s.set_microscope_parameters(100, 10, 10)
         s.axes_manager[-1].offset = 150
         s.add_elements(("B", "C"))
+        self.s = s
         self.m = s.create_model()
 
     def test_suspend_auto_fsw(self):
@@ -120,6 +121,13 @@ class TestEELSModel:
         m.enable_fine_structure()
         m.resolve_fine_structure()
         assert window == m["B_K"].fine_structure_width
+
+    def test_disable_fine_structure(self):
+        self.m.components.C_K.fine_structure_active = True
+        self.m.components.B_K.fine_structure_active = True
+        self.m.disable_fine_structure()
+        assert not self.m.components.C_K.fine_structure_active
+        assert not self.m.components.B_K.fine_structure_active
 
     def test_get_first_ionization_edge_energy_C_B(self):
         assert (self.m._get_first_ionization_edge_energy() ==
@@ -190,6 +198,112 @@ class TestEELSModel:
         self.m.remove(comp)
         assert len(self.m) == 2
 
+    def test_fit_wrong_kind(self):
+        with pytest.raises(ValueError):
+            self.m.fit(kind="wrongkind")
+
+    def test_enable_background(self):
+        self.m.components.PowerLaw.active = False
+        self.m.enable_background()
+        assert self.m.components.PowerLaw.active
+
+    def test_disable_background(self):
+        self.m.components.PowerLaw.active = True
+        self.m.disable_background()
+        assert not self.m.components.PowerLaw.active
+
+    def test_signal1d_property(self):
+        assert self.s == self.m.signal1D
+        s_new = hs.signals.EELSSpectrum(np.ones(200))
+        s_new.set_microscope_parameters(100, 10, 10)
+        self.m.signal1D = s_new
+        assert self.m.signal1D == s_new
+
+    def test_signal1d_property_wrong_value_setter(self):
+        m = self.m
+        s = hs.signals.Signal1D(np.ones(200))
+        with pytest.raises(ValueError):
+            self.m.signal1D = s
+
+    def test_remove(self):
+        m = self.m
+        c_k = m.components.C_K
+        assert c_k in m
+        m.remove(c_k)
+        assert not (c_k in m)
+
+    def test_quantify(self):
+        self.m.quantify()
+
+    def test_enable_edges(self):
+        m = self.m
+        m.components.B_K.active = False
+        m.components.C_K.active = False
+        m.enable_edges(edges_list=[m.components.B_K])
+        assert m.components.B_K.active
+        assert not m.components.C_K.active
+        m.enable_edges()
+        assert m.components.B_K.active
+        assert m.components.C_K.active
+
+    def test_disable_edges(self):
+        m = self.m
+        m.components.B_K.active = True
+        m.components.C_K.active = True
+        m.disable_edges(edges_list=[m.components.B_K])
+        assert not m.components.B_K.active
+        assert m.components.C_K.active
+        m.disable_edges()
+        assert not m.components.B_K.active
+        assert not m.components.C_K.active
+
+    def test_set_all_edges_intensities_positive(self):
+        m = self.m
+        m.components.B_K.intensity.ext_force_positive = False
+        m.components.B_K.intensity.ext_bounded = False
+        m.components.C_K.intensity.ext_force_positive = False
+        m.components.C_K.intensity.ext_bounded = False
+        m.set_all_edges_intensities_positive()
+        assert m.components.B_K.intensity.ext_force_positive
+        assert m.components.B_K.intensity.ext_bounded
+        assert m.components.C_K.intensity.ext_force_positive
+        assert m.components.C_K.intensity.ext_bounded
+
+    def test_unset_all_edges_intensities_positive(self):
+        m = self.m
+        m.components.B_K.intensity.ext_force_positive = True
+        m.components.B_K.intensity.ext_bounded = True
+        m.components.C_K.intensity.ext_force_positive = True
+        m.components.C_K.intensity.ext_bounded = True
+        m.unset_all_edges_intensities_positive()
+        assert not m.components.B_K.intensity.ext_force_positive
+        assert not m.components.B_K.intensity.ext_bounded
+        assert not m.components.C_K.intensity.ext_force_positive
+        assert not m.components.C_K.intensity.ext_bounded
+
+    def test_fix_edges(self):
+        m = self.m
+        m.components.B_K.onset_energy.free = True
+        m.components.B_K.intensity.free = True
+        m.components.B_K.fine_structure_coeff.free = True
+        m.components.C_K.onset_energy.free = True
+        m.components.C_K.intensity.free = True
+        m.components.C_K.fine_structure_coeff.free = True
+        m.fix_edges(edges_list=[m.components.B_K])
+        assert not m.components.B_K.onset_energy.free
+        assert not m.components.B_K.intensity.free
+        assert not m.components.B_K.fine_structure_coeff.free
+        assert m.components.C_K.onset_energy.free
+        assert m.components.C_K.intensity.free
+        assert m.components.C_K.fine_structure_coeff.free
+        m.fix_edges()
+        assert not m.components.B_K.onset_energy.free
+        assert not m.components.B_K.intensity.free
+        assert not m.components.B_K.fine_structure_coeff.free
+        assert not m.components.C_K.onset_energy.free
+        assert not m.components.C_K.intensity.free
+        assert not m.components.C_K.fine_structure_coeff.free
+
     def test_free_edges(self):
         m = self.m
         m.components.B_K.onset_energy.free = False
@@ -211,6 +325,28 @@ class TestEELSModel:
         assert m.components.B_K.fine_structure_coeff.free
         assert m.components.C_K.onset_energy.free
         assert m.components.C_K.intensity.free
+        assert m.components.C_K.fine_structure_coeff.free
+
+    def test_fix_fine_structure(self):
+        m = self.m
+        m.components.B_K.fine_structure_coeff.free = True
+        m.components.C_K.fine_structure_coeff.free = True
+        m.fix_fine_structure(edges_list=[m.components.B_K])
+        assert not m.components.B_K.fine_structure_coeff.free
+        assert m.components.C_K.fine_structure_coeff.free
+        m.fix_fine_structure()
+        assert not m.components.B_K.fine_structure_coeff.free
+        assert not m.components.C_K.fine_structure_coeff.free
+
+    def test_free_fine_structure(self):
+        m = self.m
+        m.components.B_K.fine_structure_coeff.free = False
+        m.components.C_K.fine_structure_coeff.free = False
+        m.free_fine_structure(edges_list=[m.components.B_K])
+        assert m.components.B_K.fine_structure_coeff.free
+        assert not m.components.C_K.fine_structure_coeff.free
+        m.free_fine_structure()
+        assert m.components.B_K.fine_structure_coeff.free
         assert m.components.C_K.fine_structure_coeff.free
 
 
@@ -237,6 +373,19 @@ class TestEELSModelFitting:
         assert intensity != m.components.B_K.intensity.value
         assert onset_energy != m.components.B_K.onset_energy.value
         assert fine_structure_coeff != m.components.B_K.fine_structure_coeff.value
+
+    def test_fix_edges(self):
+        m = self.m
+        m.enable_fine_structure()
+        intensity = m.components.B_K.intensity.value
+        onset_energy = m.components.B_K.onset_energy.value
+        fine_structure_coeff = m.components.B_K.fine_structure_coeff.value
+        m.free_edges()
+        m.fix_edges()
+        m.multifit()
+        assert intensity == m.components.B_K.intensity.value
+        assert onset_energy == m.components.B_K.onset_energy.value
+        assert fine_structure_coeff == m.components.B_K.fine_structure_coeff.value
 
 
 @lazifyTestClass
@@ -277,3 +426,24 @@ class TestFitBackground:
                         2.14)
         assert not self.m["B_K"].active
         assert not self.m["C_K"].active
+
+
+@lazifyTestClass
+class TestFitBackground2D:
+
+    def setup_method(self):
+        pl = hs.model.components1D.PowerLaw()
+        data = np.empty((2, 250))
+        data[0] = pl.function(np.arange(150, 400))
+        pl.r.value = 1.5
+        data[1] = pl.function(np.arange(150, 400))
+        s = hs.signals.EELSSpectrum(data)
+        s.set_microscope_parameters(100, 10, 10)
+        s.axes_manager[-1].offset = 150
+        self.s = s
+        self.m = s.create_model()
+
+    def test_only_current_false(self):
+        self.m.fit_background(only_current=False)
+        residual = self.s - self.m.as_signal()
+        assert pytest.approx(residual.data) == 0

--- a/hyperspy/tests/model/test_eelsmodel.py
+++ b/hyperspy/tests/model/test_eelsmodel.py
@@ -190,6 +190,54 @@ class TestEELSModel:
         self.m.remove(comp)
         assert len(self.m) == 2
 
+    def test_free_edges(self):
+        m = self.m
+        m.components.B_K.onset_energy.free = False
+        m.components.B_K.intensity.free = False
+        m.components.B_K.fine_structure_coeff.free = False
+        m.components.C_K.onset_energy.free = False
+        m.components.C_K.intensity.free = False
+        m.components.C_K.fine_structure_coeff.free = False
+        m.free_edges(edges_list=[m.components.B_K])
+        assert m.components.B_K.onset_energy.free
+        assert m.components.B_K.intensity.free
+        assert m.components.B_K.fine_structure_coeff.free
+        assert not m.components.C_K.onset_energy.free
+        assert not m.components.C_K.intensity.free
+        assert not m.components.C_K.fine_structure_coeff.free
+        m.free_edges()
+        assert m.components.B_K.onset_energy.free
+        assert m.components.B_K.intensity.free
+        assert m.components.B_K.fine_structure_coeff.free
+        assert m.components.C_K.onset_energy.free
+        assert m.components.C_K.intensity.free
+        assert m.components.C_K.fine_structure_coeff.free
+
+
+@lazifyTestClass
+class TestEELSModelFitting:
+
+    def setup_method(self, method):
+        data = np.zeros(200)
+        data[25:] = 100
+        s = hs.signals.EELSSpectrum(data)
+        s.set_microscope_parameters(100, 10, 10)
+        s.axes_manager[-1].offset = 150
+        s.add_elements(("B", ))
+        self.m = s.create_model(auto_background=False)
+
+    def test_free_edges(self):
+        m = self.m
+        m.enable_fine_structure()
+        intensity = m.components.B_K.intensity.value
+        onset_energy = m.components.B_K.onset_energy.value
+        fine_structure_coeff = m.components.B_K.fine_structure_coeff.value
+        m.free_edges()
+        m.multifit()
+        assert intensity != m.components.B_K.intensity.value
+        assert onset_energy != m.components.B_K.onset_energy.value
+        assert fine_structure_coeff != m.components.B_K.fine_structure_coeff.value
+
 
 @lazifyTestClass
 class TestFitBackground:

